internal/services/datafactory/data_factory_dataset_cosmosdb_mongoapi_resource.go

```diff
@@ -0,0 +1,339 @@
+package datafactory
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/datafactory/2018-06-01/factories"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/jackofallops/kermit/sdk/datafactory/2018-06-01/datafactory"
+)
+
+var _ sdk.Resource = DataFactoryDatasetCosmosDbMongoDbResource{}
+
+type DataFactoryDatasetCosmosDbMongoDbResource struct{}
+
+type DataFactoryDatasetCosmosDbMongoDbResourceModel struct {
+	AdditionalProperties map[string]interface{} `tfschema:"additional_properties"`
+	Annotations          []string               `tfschema:"annotations"`
+	CollectionName       string                 `tfschema:"collection_name"`
+	DataFactoryId        string                 `tfschema:"data_factory_id"`
+	Description          string                 `tfschema:"description"`
+	Folder               string                 `tfschema:"folder"`
+	LinkedServiceName    string                 `tfschema:"linked_service_name"`
+	Name                 string                 `tfschema:"name"`
+	Parameters           map[string]interface{} `tfschema:"parameters"`
+}
+
+func (DataFactoryDatasetCosmosDbMongoDbResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"additional_properties": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"annotations": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"collection_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"data_factory_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: factories.ValidateFactoryID,
+		},
+		"description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"folder": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"linked_service_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"name": {
```

Comment:
Can we reorder the schema here? Conventionally, the `name` and any parent resource name attributes come first, then those required ones (in alphabetic order), then the optional ones (alphabetically), then the computed ones.

Reply:
Fixed in 1179a4200a